### PR TITLE
EZP-25859: Add loadDrafts method to User model

### PR DESCRIPTION
**Pull request by sunpietro**
https://jira.ez.no/browse/EZP-25859

**TO DO:**
- [x] added `loadDrafts` method to `Y.eZ.User` model definition,
- [x] updated JS unit tests

Requires: https://jira.ez.no/browse/EZP-25753

### CODE REVIEW

**Comment by sunpietro** [comment]
ping @dpobel @StephaneDiot @yannickroger

File: Resources/public/js/models/ez-usermodel.js [hunk 1]
```diff
@@ -72,7 +72,37 @@ YUI.add('ez-usermodel', function (Y) {
...
-        }
+        },
+
+        /**
+         * Loads drafts created by a given user
+         *
+         * @method loadDrafts
+         * @param options {Object}
+         * @param options.api {Object} (required) the JS REST client instance
+         * @param userId {String} the REST user id
```
Comment (dpobel) [inline Resources/public/js/models/ez-usermodel.js]: wrong parameter

File: Resources/public/js/models/ez-usermodel.js [hunk 2]
```diff
@@ -72,7 +72,37 @@ YUI.add('ez-usermodel', function (Y) {
...
+         * @param callback {Function} function to call after processing response
+         */
+        loadDrafts: function (options, callback) {
+            var versions = [];
```
Comment (dpobel) [inline Resources/public/js/models/ez-usermodel.js]: nitpick: should be moved to the loadUserDrafts callback

File: Tests/js/models/assets/ez-usermodel-tests.js [hunk 3]
```diff
@@ -188,7 +191,107 @@ YUI.add('ez-usermodel-tests', function (Y) {
...
+                        "names":{"value":[{"_languageCode":"eng-GB","#text":"2 zones"}]},
+                        "Content":{"_href":"/api/ezp/v2/content/objects/116"}
+                    }
+                },
+                loadUserDraftsResponse = {document: {VersionList: {VersionItem: [draftHash]}}},
+                loadDraftsCallback = function (error, response) {
+                    isCallbackCalled = true;
+
+                    Y.Assert.isFalse(error, 'Should not throw error');
+                    Y.Assert.areSame(1, response.length, 'Should return one draft model');
```
Comment (dpobel) [inline Tests/js/models/assets/ez-usermodel-tests.js]: instead of `1`, you should use `loadUserDraftsResponse.document.VersionList.VersionItem.length`

File: Tests/js/models/assets/ez-usermodel-tests.js [hunk 4]
```diff
@@ -188,7 +191,107 @@ YUI.add('ez-usermodel-tests', function (Y) {
...
+
+            Y.Mock.expect(contentServiceMock, {
+                method: 'loadUserDrafts',
+                args: [this.modelId, Y.Mock.Value.Function],
+                run: function (id, callback) {
+                    callback(false, loadUserDraftsResponse);
+                }
+            });
+
+            Y.Mock.expect(capiMock, {
```
Comment (dpobel) [inline Tests/js/models/assets/ez-usermodel-tests.js]: to avoid repeating that in each test, the CAPI mock is usually defined in `setUp`

File: Tests/js/models/ez-usermodel.html [hunk 5]
```diff
@@ -10,26 +11,27 @@
 <script type="text/javascript" src="./assets/ez-usermodel-tests.js"></script>
 <script>
     var filter = (window.location.search.match(/[?&]filter=([^&]+)/) || [])[1] || 'raw',
-        loaderFilter;
-    if (filter == 'coverage'){
-        loaderFilter = {
+        loaderFilter = (filter !== 'coverage') ? filter : {
```
Comment (dpobel) [inline Tests/js/models/ez-usermodel.html]: can you please stop doing unrelated changes ? This is completely unrelated to the addition of the `loadDrafts` method (not to mention the very limited added value).

**Comment by sunpietro** [comment]
ping @dpobel

**Comment by dpobel** [comment]
+1

**Comment by andrerom** [comment]
+1

**Comment by dpobel** [comment]
aaah I've just realized you created the PR against master not against the dashboard feature branch :( @sunpietro  Please do the dashboard related changes on that branch.
Anyway, I'm rebasing the feature branch against master.

**Comment by sunpietro** [comment]
I felt it should go to master instead of feature branch. 
@dpobel what did you mean by: `Please do the dashboard related changes on that branch.`?

**Comment by dpobel** [comment]
@sunpietro at the moment, the loadUserDrafts method is only needed by the corresponding dashboard block that's why this PR should have been created against `feature/ezp-25766_dashboard` not `master`, this would have avoid some extras rebase operations for both you and me :)

**Comment by sunpietro** [comment]
@dpobel thanks for clarification, I'll try to avoid it in the future.
